docs/tutorial_basics.rst

```diff
@@ -429,17 +429,30 @@ Here is the corresponding ReFrame test, where the new concepts introduced are hi
    :lines: 6-
    :emphasize-lines: 11-13
 
+
+In order to compile applications using ``std::thread`` with GCC and Clang, the ``-pthread`` option has to be passed to the compiler.
+Since the above option might not be valid for other compilers, we use pipeline hooks to differentiate based on the programming environment as follows:
+
+.. code-block:: python
+
+   @rfm.run_before('compile')
+   def setpthreadsflag(self):
+       environ = self.current_environ.name
+       if environ in {'clang', 'gnu'}:
+           self.build_system.cxxflags += ['-pthread']
+
+
 .. note::
 
-   The ``-pthread`` option is needed to compile applications using ``std::thread`` with GCC and Clang.
-   In case of a different compiler, the above option may need to be adjusted.
+   The pipeline hooks are covered in more detail in a following tutorial section.
+   For a detailed coverage of the regression test pipeline, please refer to :doc:`pipeline`.
```

Comment:
I wouldn't put this sentence here. The pipeline is covered in the discussion about the hooks.